sklearn/preprocessing/tests/test_data.py

```diff
@@ -1353,20 +1353,57 @@ def test_quantile_transform_subsampling():
     X = sparse.rand(n_samples, 1, density=0.99, format="csc", random_state=0)
     inf_norm_arr = []
     for random_state in range(ROUND):
-        transformer = QuantileTransformer(
+        transformer_s0 = QuantileTransformer(
             random_state=random_state,
             n_quantiles=n_quantiles,
             subsample=n_samples // 10,
         )
-        transformer.fit(X)
-        diff = np.linspace(0, 1, n_quantiles) - np.ravel(transformer.quantiles_)
+        transformer_s0.fit(X)
+        diff = np.linspace(0, 1, n_quantiles) - np.ravel(transformer_s0.quantiles_)
         inf_norm = np.max(np.abs(diff))
         assert inf_norm < 1e-1
         inf_norm_arr.append(inf_norm)
     # each random subsampling yield a unique approximation to the expected
     # linspace CDF
     assert len(np.unique(inf_norm_arr)) == len(inf_norm_arr)
 
+    # check subsampling disabled
```

Comment:
I prefer to add a new dedicated function that check the implementation of `subsample=None`.

While you will need to add a description via the docstring of the test, we don't expect any comment in the code since it should be as much as possible self explicit.

I assume that to test this feature, I would expect the test to check that we don't have any clipping on the limit min/max instead of checking all the quantiles.

Reply:
I pulled out the test and moved the comments to the docstring.

Regarding checking all the all the quantiles vs the min/max, I would argue we should test all the quantiles as more than the min/max will change when the sample is different. The computational costs of comparing all the quantiles is negligible and if we're worried about the floating noise causing false positive tests when adjust the tolerances.